docs/inference_api_beta/README.md

```diff
@@ -0,0 +1,297 @@
+# Gretel Tabular LLM Inference API (Beta)
+
+This README details the API that powers the [Gretel Model Playground](https://console.gretel.ai/playground).  This API is in **Beta** and subject to change at any time. If you have used the Model Playground and are interested in building your own application with Gretel, then you may uitlize the API directly based on the documentation below. 
+
+Within this directory there are examples with Javascript and Python that provide lighweight wrappers around the API.
+
+# General Operation
+
+Gretel Tabular LLM utilizes lightweight streams to generate and retrieve tabular data. Compared to other LLM streaming mechanisms, such as Server-Sent Events (SSE), Gretel does not stream back individual tokens. The Tabular LLM infrastructure automatically utilizes multiple LLMs and Agents to generate tokens and re-assemble them into tabular data. Client-side applications can then poll specific REST API endpoints to retrieve table rows as they are generated and assembled. By using a lightweight polling mechanism, Tabular LLM can be more flexibly integrated into apps that may not have streaming capabilities.
+
+There are two main steps to utilize the Tabular LLM APIs:
+
+1) First, a request is made that contains the prompt and LLM parameters. This request will return immediately with a *inference stream ID*. This ID represents a lightweight stream that can be itereated next. Behind the scenes, this request will be processed and tabular records will be generated and written to the stream.
+
+2) Second, API calls are made to iterate the *inference stream* to read tabular records (in JSON format) until generation is complete.
+
+All API calls are made using `HTTP POST` requests.
+
+## Authentication
+
+In order to use this API you will need a [Gretel API key](https://console.gretel.ai/users/me/key). This API key will need to be included in all API call [HTTP headers](https://api.docs.gretel.ai/). If you are using the Javascript examples, you can set the API key as an enviroment variable. If you are using the Python examples, we highly suggest you install the [Gretel Python SDK](https://docs.gretel.ai/guides/environment-setup/cli-and-sdk) and run `gretel configure` to configure API authentication.
+
+## Rate Limiting
+
+Stream creation (LLM requests) are rate limited to approximately 3 requests every 15 seconds. On average, you may create a stream every 5 seconds. If this limit is exceeded, the stream creation endpoint will return HTTP 403 errors.
+
+# API Usage
+
+These sections details the specifics of making API calls for Tabular LLM. All requests are made with `HTTP POST` requests.
+
+## Make a LLM Request
+
+To submit a prompt to Tabular LLM, make the following call. This example uses one of the sample prompts from the Model Playground. Please update the prompt to suit your needs.
+
+```
+POST https://api.gretel.cloud/v1/inference/tabular/stream
+
+{
+  "model_id": "gretelai/tabular-v0",
```

Comment:
is it worth calling out the `/models` endpoint? or do we just want to direct people to `tabular-v0` anyways, and unless you are a partner, you have no idea what the difference between a/b/c is

Reply:
Yeah I left that part out for now. We can add it if we need too. There's a helper function that pulls the model IDs, too.